ssz/sedes/serializable.py

```diff
@@ -162,7 +167,7 @@ def __hash__(self):
             self._hash_cache = hash(self.__class__) * int.from_bytes(self.hash_tree_root, "little")
         return self._hash_cache
 
-    def copy(self, *args, **kwargs):
+    def copy(self, clear_cache=False, *args, **kwargs):
```

Comment:
This strikes me as somewhat surprising behavior.  IIUC this cache clearing would extend beyond this individual object  to all objects using this database?  That strikes me as something that shouldn't occur as part of an individual object copy operation but something that would be directly performed on the cache itself.

Reply:
I rework it to [`SSZCache`](https://github.com/ethereum/py-ssz/blob/3000cf09661d6ab67980442a0fda87755dc807fa/ssz/cache/cache.py), it only exists inside `Serializable` object.

The next plan is to make the `Serializable.__init__` accept `db` object, to store the **field values**. (SSZCache is only for Merkle leaves currently)